GVFS/GVFS.Common/ProductUpgraderInfo.cs

```diff
@@ -16,24 +16,39 @@ public ProductUpgraderInfo(ITracer tracer, PhysicalFileSystem fileSystem)
             this.fileSystem = fileSystem;
         }
 
+        public static ProductUpgraderInfoImpl GetProductUpgraderInfoImpl()

```

Comment:
This just doesn't feel right to me.  Especially when you are calling this method for every call to get a directory?  Can we have an interface and create the instance that implements it only once and pass around the interface where we need it and get rid of all the statics?

Reply:
I like the final goal here - I can at least avoid creating a new instance for each call.